packages/world/src/modules/init/implementations/WorldRegistrationSystem.sol

```diff
@@ -264,27 +265,7 @@ contract WorldRegistrationSystem is System, IWorldErrors, LimitedCallContext {
     ResourceId delegationControlId,
     bytes memory initCallData
   ) public onlyDelegatecall {
-    // Store the delegation control contract address
-    UserDelegationControl._set({
-      delegator: _msgSender(),
-      delegatee: delegatee,
-      delegationControlId: delegationControlId
-    });
-
-    // If the delegation is limited...
-    if (Delegation.isLimited(delegationControlId)) {
-      // Require the delegationControl contract to implement the IDelegationControl interface
-      address delegationControl = Systems._getSystem(delegationControlId);
-      requireInterface(delegationControl, type(IDelegationControl).interfaceId);
-
-      // Call the delegation control contract's init function
-      SystemCall.callWithHooksOrRevert({
-        caller: _msgSender(),
-        systemId: delegationControlId,
-        callData: initCallData,
-        value: 0
-      });
-    }
+    createDelegation(_msgSender(), delegatee, delegationControlId, initCallData);
```

Comment:
what do you think about calling this helper function `registerDelegation`?

Reply:
I considered that, but it might be confusing and because this is internal, a godlike "create" makes sense to me.

Plus the compiler doesn't let you! 
<img width="691" alt="image" src="https://github.com/latticexyz/mud/assets/29184158/ae1db105-b866-40d7-8525-6ebaebe693c0">